OPHD/States/MapViewStateIO.cpp

```diff
@@ -79,6 +79,17 @@ static void readRccRobots(NAS2D::Xml::XmlAttribute* attr, Structure& structure,
  * CLASS FUNCTIONS
  *****************************************************************************/
 
+void MapViewState::del(const std::string& filePath) {
+	if (Utility<Filesystem>::get().exists(filePath))
+	{
+		Utility<Filesystem>::get().del(filePath);
+	}
+	else
+	{
+		throw std::runtime_error("File '" + filePath + "' was not found.");
+	}
+}
```

Comment:
I'm wondering if this functionality should be part of the `FileIo` dialog, rather than `MapViewState`.

----

This function contains a time-of-check to time-of-use race condition. It's possible the file can be deleted by another process between the calls to `exists` and `del`.

Since `Filesystem::del` will already `throw` if the file doesn't exist, it may be better to just call `Filesystem::del`, and let the exception handling there raise an exception when the file doesn't exist.

Reply:
To add a bit of clarity, the method could be simplified to:
```cpp
void MapViewState::del(const std::string& filePath) {
	Utility<Filesystem>::get().del(filePath);
}
```

Which of course means we don't really need the method, and could just call: `Utility<Filesystem>::get().del(filePath);`

Internally the `Filesystem::del` method tries to delete the file, and if it didn't exist, it will `throw` and exception. Thus there is little need to detect this situation ahead of time, and write a custom `throw`.

```cpp
void Filesystem::del(const std::string& filename) const
{
	if (PHYSFS_delete(filename.c_str()) == 0)
	{
		throw std::runtime_error(std::string("Unable to delete '") + filename + "':" + PHYSFS_getErrorByCode(PHYSFS_getLastErrorCode()));
	}
}
```

The race condition is avoided because the `PHYSFS_delete` method combines the delete and existence check into a single atomic operation. If the file exists, it's deleted. If it doesn't exist, the return value indicates the file didn't exist, and that error return value gets translated into an exception.